### DISCUSSION TITLE
How to propagate correctly tags into Opsgenie

### DISCUSSION THREAD

**Discussion by rschirin** [jertel/elastalert2#534] [Q&A] [answered]
Hey there,
I'm trying to integrate Elastalert2 with Opsgenie. Everything is fine but the propagation of tags from heartbeat rule to Opsgenie alert.
Basically, the first alert is created with correct tags using the configuration below:

```
opsgenie_tags:
   - "{tags[0]}"
   - "{tags[1]}"
   - "{monitor[id]}"
```

while next one maintains the same values even if they are different.
How should I fix that?

**Comment by jertel** [comment, marked as answer]
Looks like a bug on this line: https://github.com/jertel/elastalert2/blob/93ee6ff60f94406dff9ce57ba7b0c7c67efdf0f7/elastalert/alerters/opsgenie.py#L91

It's formatting the tags on the first alert and saving the newly formatted tags over the top of the original configured tags, instead of saving the formatted tag into a temporary variable. Here's the offending commit: https://github.com/jertel/elastalert2/commit/386708e223329f9ececa9e0f2df4d3b60acad3c6

It should be a simple fix if you're up for a PR. If you're unable then someone may eventually get to it.

**Reply by nsano-rururu** [reply]
Is the following pull request on yelp / elastalert a relevant fix?

Fix Incorrect Opsgenie Tags Formatting
https://github.com/Yelp/elastalert/pull/2884
Patch opsgenie tags overwritten after first alert 
https://github.com/Yelp/elastalert/pull/2840

**Reply by rschirin** [reply]
So, is it possible to use those pull requests or should be created a new one?

**Reply by jertel** [reply]
Fixed in #537.
